armi/reactor/blueprints/tests/test_customIsotopics.py

```diff
@@ -216,18 +332,89 @@ def test_massFractionsAreApplied(self):
             :id: T_ARMI_MAT_USER_INPUT3
             :tests: R_ARMI_MAT_USER_INPUT
         """
-        fuel0 = self.a[0].getComponent(Flags.FUEL)
         fuel1 = self.a[1].getComponent(Flags.FUEL)
         fuel2 = self.a[2].getComponent(Flags.FUEL)
         self.assertEqual(self.numCustomNuclides, len(fuel1.p.numberDensities))
         self.assertAlmostEqual(19.1, fuel1.density())
 
-        # density only works with a Custom material type.
-        self.assertAlmostEqual(fuel0.density(), fuel2.density())
         self.assertEqual(
             set(fuel2.p.numberDensities.keys()), set(fuel1.p.numberDensities.keys())
         )  # keys are same
 
+    def test_densitiesAppliedToNonCustomMaterials(self):
+        """Ensure that a density can be set in custom isotopics for components using library materials."""
+        # The template block
+        fuel0 = self.a[0].getComponent(Flags.FUEL)
+        # The block with custom density but not the 'Custom' material
+        fuel2 = self.a[2].getComponent(Flags.FUEL)
+        # A block like the template block, but made after the custom block
+        fuel6 = self.a[6].getComponent(Flags.FUEL)
+        # A block with custom density set via number density
+        fuel8 = self.a[8].getComponent(Flags.FUEL)
+
+        # Check that the density is set correctly on the custom density block,
+        # and that it is not the same as the original
+        self.assertAlmostEqual(19.1, fuel2.density())
```

Comment:
If you are asking if the custom density logic should be adjusting to the working temperature, probably. When I wrote this PR, I had misunderstood what Tin in the blueprints was used for, see https://github.com/terrapower/armi/issues/1818#issuecomment-2289766135 for more details. In short, there is an assumption of hot heights in the blueprints unless the option `inputHeightsConsideredHot` is `False`, so in the default case of `inputHeightsConsideredHot`=True the feature in this PR will assign incorrect densities for components with custom isotopics because this feature assumes the blueprints are for cold dimensions.